test/ntttcp_test.py

```diff
@@ -0,0 +1,441 @@
+import pytest
+import subprocess
+from typing import Optional, Tuple
+from xml.sax import make_parser
+from xml.sax.handler import ContentHandler
+
+class NtttcpResult:
+    def __init__(
+        self,
+        rec_out : str,
+        send_out : str,
+        ) -> None:
+        self.rec_stdout = rec_out
+        self.send_stdout = send_out
+
+class ParseNtttcpResult:
+    throughput_pattern = re.compile(
+        r"INFO: \\t throughput\\t:(\d+.\d+)Gbps"
+    )  # need to refine, for the throughput is not always Gbps
+    daemon_pattern = re.compile(r"\*\*\*\s+run as daemon")
+    multi_clients_pattern = re.compile(r"multiple clients:\s+yes")
+    epoll_pattern = re.compile(r"\*\*\*\s+use epoll()")
+    system_tcp_retransmit_pattern = re.compile(r"show system tcp retransmit:\s+yes")
+
+    def __init__(self, rec_out: str, send_out: str) -> None:
+        self.rec_stdout = rec_out
+        self.send_stdout = send_out
+
+    #13:38:16 INFO:   throughput     :20.68Gbps
+    def get_throughput(self) -> float:
+        throughput = 0
+        matched = self.throughput_pattern.findall(str(self.send_stdout))
+        if matched:
+            throughput = matched[0]
+        return float(throughput)
+
+    #*** run as daemon
+    def is_daemon_running(self) -> bool:
+        if self.daemon_pattern.findall(str(self.rec_stdout)):
+            return True
+        else:
+            return False
+
+    #multiple clients:                yes
+    def is_multi_clients_mode(self) -> bool:
+        if self.multi_clients_pattern.findall(str(self.rec_stdout)):
+            return True
+        else:
+            return False
+
+    #connections:                     4 X 5 X 3
+    def is_multi_ports_threads(self, send_cmd) -> bool:
+        output_value_pattern = re.compile(r"connections:\s+(.+) X (.+) X (.+)")
+        cmd_port_pattern = re.compile(r"-P (\d+)")
+        cmd_threads_pattern = re.compile(r"-n (\d+)")
+        cmd_connections_pattern = re.compile(r"-l (\d+)")
+        output_value = output_value_pattern.findall(str(self.send_stdout))
+        cmd_port = cmd_port_pattern.findall(send_cmd)
+        cmd_threads = cmd_threads_pattern.findall(send_cmd)
+        cmd_connections = cmd_connections_pattern.findall(send_cmd)
+        listening_port = output_value[0][0]
+        threads_on_receiver = output_value[0][1]
+        connections_on_sender = output_value[0][2]
+        if cmd_port[0] == listening_port and cmd_threads[1] == threads_on_receiver and cmd_connections[0] == connections_on_sender:
+            return True
+        else:
+            return False
+
+    #test warm-up (sec):              4
+    #test duration (sec):             5
+    #test cool-down (sec):            5
+    def is_running_time(self, common_option) -> bool:
+        warmup_pattern = re.compile(r"test warm-up \(sec\):\s+(\d+)")
+        warmup_time = warmup_pattern.findall(str(self.send_stdout))
+        duration_pattern = re.compile(r"test duration \(sec\):\s+(\d+)")
```

Comment:
Also with this duration pattern. Match the result duration. 
01:21:46 INFO: test duration    :5.35 seconds

Reply:
done